### PR TITLE
Align the server and client idle timeouts in Ember

### PR DESCRIPTION
**Pull request by rlavolee**
Fixes: [#7327 ](https://github.com/http4s/http4s/issues/7327)

### CODE REVIEW

File: ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala [hunk 1]
```diff
@@ -429,8 +429,8 @@ object EmberClientBuilder extends EmberClientBuilderCompanionPlatform {
     val acgFixedThreadPoolSize: Int = 100
     val chunkSize: Int = 32 * 1024
     val maxResponseHeaderSize: Int = 4096
-    val idleConnectionTime: FiniteDuration = org.http4s.client.defaults.RequestTimeout
-    val timeout: Duration = org.http4s.client.defaults.RequestTimeout
+    val idleConnectionTime: FiniteDuration = org.http4s.ember.core.Defaults.idleTimeout
+    val timeout: Duration = org.http4s.ember.core.Defaults.idleTimeout
```
Comment (danicheg) [inline ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala:433]: Hmm, maybe changing the default `timeout` is not desired. I was thinking that `45s` is quite a lot but `60s` 🤯

File: ember-core/shared/src/main/scala/org/http4s/ember/core/Defaults.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * limitations under the License.
+ */
+
+package org.http4s.ember.core
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.duration.FiniteDuration
+
+private[ember] object Defaults {
+  val idleTimeout: FiniteDuration = 60.seconds
```
Comment (danicheg) [inline ember-core/shared/src/main/scala/org/http4s/ember/core/Defaults.scala:23]: We have different notations for such vals over the codebase, but maybe PascalCase better fits. wdyt?

**Comment by danicheg** [comment]
I agree with having a single source of truth for general client/server defaults. I doubt a little about putting it into `ember-core`, but we don't have the `ember-common`, and it's unlikely worth the hassle. So it seems to be an applicable trade-off. Eager to hear opinions from the team.

File: ember-core/shared/src/main/scala/org/http4s/ember/core/Defaults.scala (hunk 2 above)
Comment (rlavolee) [reply ember-core/shared/src/main/scala/org/http4s/ember/core/Defaults.scala]: That's interesting. I saw camelCase and PascalCase vals in this project but how do you choose between eatch other ?

File: ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala (hunk 1 above)
Comment (rlavolee) [reply ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala]: My bad 😬 . I reverted the timeout from previous value.

**Review by valencik — APPROVED** [review]
Thanks @rlavolee for your efforts here.

The technical approach seems fine.
Because this is a change of behaviour, we need to remember to call it our clearly in our next release notes.

(edit: just wanted to link the brief discussion about whether this change can go in a patch release: https://github.com/http4s/http4s/issues/7327#issuecomment-1827866282)

**Comment by danicheg** [comment]
Thanks again @rlavolee!

**Pull request merged**